internal/wait/noderesourcetopologies.go

```diff
@@ -110,3 +113,65 @@ func (wt Waiter) ForNodeResourceTopologiesSettled(ctx context.Context, threshold
 	klog.Infof("done waiting for NRT to stabilize; observed nodes=%d ready=%d", pfpState.Len(), pfpState.CountReady(threshold))
 	return &nrtList, err
 }
+
+type NRTFilterFunc func(nrt *nrtv1alpha2.NodeResourceTopology) bool
+
+func NRTPassthrough(nrt *nrtv1alpha2.NodeResourceTopology) bool {
+	return true
+}
+
+func (wt Waiter) ForNodeResourceTopologiesEqualTo(ctx context.Context, nrtListReference *nrtv1alpha2.NodeResourceTopologyList, nrtFilter NRTFilterFunc) (nrtv1alpha2.NodeResourceTopologyList, error) {
+	var updatedNrtList nrtv1alpha2.NodeResourceTopologyList
+	klog.Infof("Waiting up to %v to have %d NRT objects restored", wt.PollTimeout, len(nrtListReference.Items))
+	err := k8swait.Poll(wt.PollInterval, wt.PollTimeout, func() (bool, error) {
+		err := wt.Cli.List(ctx, &updatedNrtList)
+		if err != nil {
+			klog.Errorf("cannot get the NRT List: %v", err)
+			return false, err
+		}
+		for idx := range nrtListReference.Items {
+			referenceNrt := &nrtListReference.Items[idx]
+
+			if !nrtFilter(referenceNrt) {
+				klog.Warningf("skipping NRT %q because of filter", referenceNrt.Name)
+				continue
+			}
+
+			updatedNrt, err := findNRTByName(updatedNrtList.Items, referenceNrt.Name)
+			if err != nil {
+				klog.Errorf("missing NRT for %s: %v", referenceNrt.Name, err)
+				return false, err
+			}
+			if !isNRTEqual(*referenceNrt, *updatedNrt) {
+				klog.Warningf("NRT for %s does not match reference", referenceNrt.Name)
+				return false, err
+			}
+			klog.Info("NRT for %s matches reference", referenceNrt.Name)
+		}
+		klog.Infof("Matched reference for all %d NRT objects", len(nrtListReference.Items))
+		return true, nil
+	})
+	return updatedNrtList, err
+}
+
+func findNRTByName(nrts []nrtv1alpha2.NodeResourceTopology, name string) (*nrtv1alpha2.NodeResourceTopology, error) {
+	for idx := 0; idx < len(nrts); idx++ {
+		if nrts[idx].Name == name {
+			return &nrts[idx], nil
+		}
+	}
+	return nil, fmt.Errorf("failed to find NRT for %q", name)
+}
+
+func isNRTEqual(initialNrt, updatedNrt nrtv1alpha2.NodeResourceTopology) bool {
```

Comment:
We already have it here https://github.com/openshift-kni/numaresources-operator/pull/608/commits/ee9fa855df1a0ab9d6a91dd4f0a38ea271c7086b#diff-98853065ac58ab9f346c413d9e8a25d4268fc59f11d0012dc087d28b09798e81R107 don't we?

Reply:
we do, but this package can't depend on `test/utlls`, while the other way around is allowed

Reply:
So why not having it the other way around?

Reply:
because `IsEqualNRT` and `FindNRTByName` don't really fit in the `wait` package. They belong more to a package dealing with NRT objects rather than a package dealing with wait for conditions.
Lacking (time for) a refactoring to repurpose and reshuffle the code, [a little duplication seems better than a little dependency](https://www.efekarakus.com/2021/09/23/a-little-copying-is-better-than-a-little-dependency.html)

Reply:
I see, let's move on, it's not a big deal anyway